pkg/sql/row/kv_fetcher.go

```diff
@@ -210,9 +210,15 @@ func (f *BackupSSTKVFetcher) nextBatch(
 			}
 		}
 		if len(f.iter.UnsafeValue()) == 0 {
-			// Value is deleted.
-			f.iter.NextKey()
-			continue
+			if f.endTime.IsEmpty() || f.iter.UnsafeKey().Timestamp.Less(f.endTime) {
```

Comment:
This change either:

- does not belong to this PR, or
- needs a unit test, AND a review from the KV and SQL queries team.

Reply:
Yep, this change does not belong to this PR. Backup related tests were in ccl pkg. Not sure if we can import ccl from sql pkg. So I made the unit tests in cliccl/debug_backup_test.go.